Library/Homebrew/extend/os/linux/install.rb

```diff
@@ -76,23 +76,43 @@ def symlink_ld_so
     end
     private_class_method :symlink_ld_so
 
-    def symlink_gcc_libs
+    def setup_preferred_gcc_libs
       gcc_opt_prefix = HOMEBREW_PREFIX/"opt/#{OS::LINUX_PREFERRED_GCC_FORMULA}"
+      ld_so = HOMEBREW_PREFIX/"opt/glibc/lib/ld-linux-x86-64.so.2"
+
+      if ld_so.readable?
+        ld_so_conf_d = HOMEBREW_PREFIX/"etc/ld.so.conf.d"
+        ld_so_conf_d.mkpath
+        ld_so_conf_d.chmod "go-w"
```

Comment:
`pathname.chmod` does not support mode string.

```console
$ brew irb
irb(main):001:0> p = Pathname("temp")
=> #<Pathname:temp>
irb(main):002:0> p.mkpath
=> nil
irb(main):003:0> p.chmod "go-w"
Traceback (most recent call last):
        4: from /usr/local/Homebrew/Library/Homebrew/brew.rb:95:in `<main>'
        3: from (irb):3
        2: from (irb):3:in `chmod'
        1: from (irb):3:in `chmod'
TypeError (no implicit conversion of String into Integer)
irb(main):004:0>
```

Should use `FileUtils.chmod mode, path` instead:

```suggestion
        chmod "go-w", ld_so_conf_d
```